widget/diagramwidget/geometry/r2/vec2.go

```diff
@@ -0,0 +1,90 @@
+// Package r2 implements operations relating to objects in R2.
+package r2
+
+import (
+	"math"
+)
+
+// Vec2 implements a vector in R2
+type Vec2 struct {
```

Comment:
It may not be needed to use this, there is a `fyne.Vector` created since this code started.